packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java

```diff
@@ -260,29 +261,54 @@ public ResponseEntity<UUID> equationsToModel(
 			}
 		}
 
-		// Create a request for SKEMA with the cleaned-up equations.
-		final JsonNode skemaRequest = mapper.createObjectNode().put("model", "petrinet").set("equations", equationsReq);
-
-		// Get an AMR from Skema Unified Service
+		// Get an AMR from Skema Unified Service or MIRA
 		final Model responseAMR;
-		try {
-			responseAMR = skemaUnifiedProxy.consolidatedEquationsToAMR(skemaRequest).getBody();
-			if (responseAMR == null) {
-				log.warn("Skema Unified Service did not return a valid AMR based on the provided equations");
-				throw new ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, messages.get("skema.bad-equations"));
+		String extractionService = "mira";
+		if (req.get("extractionService") != null) {
```

Comment:
It feels really incorrect to have this be strongly typed on the front end in `knowledge.ts` and then have to deal with parsing json for values on the back end. This feels like something that should be generated from the back end and used from Types.ts

Reply:
I agree with your comment, but this is a temporary solution for the hackathon, until we prove that MIRA is good enough and get rid of SKEMA.